### PR TITLE
adding community featured models

### PR DESCRIPTION
**Pull request by Ssofja**
> [!IMPORTANT]
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Adds a **Featured Community Checkpoints** documentation page that highlights external fine-tunes built on NVIDIA NeMo ASR checkpoints, with runtime-specific load instructions for NeMo, MLX, and GGUF.

**Collection**: ASR

# Changelog

- Added `docs/source/asr/featured_community_checkpoints.rst` - new page listing community checkpoints grouped by runtime:
  - **NeMo** - 4 Hugging Face models with `.nemo` weights (`akera/parakeet-tdt-salt`, `johannhartmann/parakeet_de_med`, `qenneth/parakeet-tdt-0.6b-v3-finetuned-for-ATC`, `KasuleTrevor/parakeet-0.6b-cv-sw-5hr_v9`)
  - **MLX Inference** -`NeurologyAI/neuro-parakeet-mlx` with `parakeet-mlx` usage
  - **GGUF Inference**-`cstr/parakeet-tdt-0.6b-v3-GGUF` and `cstr/canary-1b-v2-GGUF` with CrispASR build/run commands
  - **Submit a Community Checkpoint** -instructions for suggesting additions via GitHub issue
- Added the page as the last entry in the ASR collection toctree (`docs/source/asr/intro.rst`)
- Added cross-link from `docs/source/asr/asr_checkpoints.rst` to the new community page
- Added cross-link from `docs/source/starthere/choosing_a_model.rst` under "Where to Find Models"

# Usage

```python
import nemo.collections.asr as nemo_asr

model = nemo_asr.models.ASRModel.from_pretrained("johannhartmann/parakeet_de_med")
print(model.transcribe(["audio.wav"])[0].text)
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Docs-only change; no code or test changes required
- Community checkpoints are curated examples, not NVIDIA-maintained models
- Page path: `asr/featured_community_checkpoints.html`

### CODE REVIEW

File: docs/source/asr/featured_community_checkpoints.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+     - CC-BY-4.0
+     - German medical/neurology ASR for Apple Silicon. WER 1.04% on the author's medical validation set.
+
+
+.. _gguf-inference:
+
+GGUF Inference
+--------------
+
+GGUF exports run with the `CrispASR <https://github.com/CrispStrobe/CrispASR>`_ C++ CLIs — no NeMo install required:
```
Comment (pzelasko) [inline docs/source/asr/featured_community_checkpoints.rst]: I think we shouldn't aim to provide up-to-date installation instructions for these frameworks. Can we re-arrange this page to be just one table that says sth like:

ckpt name | what's special about it | framework | url

Comment (Ssofja) [reply docs/source/asr/featured_community_checkpoints.rst]: Done, check please

**Review by pzelasko — APPROVED** [review]
